---
title: "TCGA survival analysis"
author: "Mikhail Dozmorov"
date: "`r Sys.Date()`"
output:
  pdf_document:
    toc: no
  html_document:
    theme: united
    toc: no
csl: styles.ref/genomebiology.csl
bibliography: data.TCGA/TCGA.bib
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```

```{r settings}
n_max  <- 10 # Maximum number of rows to output in tables
gene   <- "SDC1" # Search and replace the name of the gene
cancer <- "BRCA"
```

# Analysis 2: `r gene` in all cancers

The barplot shows the significance of `r gene` expression on survival in a given cancer. The wider (higher) the bar the more significant survival effect the gene has. See abbreviations of cancer types at [http://www.liuzlab.org/TCGA2STAT/CancerDataChecklist.pdf](http://www.liuzlab.org/TCGA2STAT/CancerDataChecklist.pdf)   

\begin{center}
\includegraphics[height=300px]{res.genes.Analysis2/SDC1_all_TCGA_cancers.png}
\end{center}

The same data in table format. Legend:

- `Cancer`, `Cancer.Name` - cancer abbreviation and description
- `Gene` - gene name for which survival analysis was run
- `p.value` - significance of the survival effect
- `HR`, `HR_left`, `HR_right` - hazard ratio, and left/right confidence interval
- `Min.`, `X1st.Qu.`, `Median`, `Mean`, `X3rd.Qu.`, `Max.` - expression level of the gene in a corresponding cancer
- `Cutoff_type`, `Cutoff_value` - gene expression cutoff best discriminating survival

```{r}
cancers <- openxlsx::read.xlsx("data.TCGA/TCGA_cancers.xlsx")
mtx <- read.table("res.genes.Analysis2/global_stats.txt", sep = "\t", header = TRUE)
mtx <- mtx[order(mtx$p.value), ]
mtx <- left_join(mtx, cancers, by = c("Cancer" = "Acronym"))
mtx <- mtx[, c("Cancer", "Cancer.Name", "Gene", "p.value", "HR", "HR_left", "HR_right", "Min.", "X1st.Qu.", "Median", "Mean", "X3rd.Qu.", "Max.", "Cutoff_type", "Cutoff_value")]
# DT::datatable(mtx)
kable(mtx[1:min(nrow(mtx), n_max), ])
```

## Survival effect in `r cancer` cancer

\begin{center}
\includegraphics[height=250px]{res.genes.Analysis2/SDC1_BRCA.png}
\end{center}

\pagebreak

# Analysis 3: `r gene` in `r cancer`, clinical subtypes

The table lists clinical subtypes where the expression of `r gene` gene in `r cancer` most significantly affects survival. The table is sorted by increasing p-values, most significant on top. Description of clinical subtypes can be found at [https://gdc.cancer.gov/about-data/data-harmonization-and-generation/clinical-data-harmonization](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/clinical-data-harmonization)

```{r}
global_stats <- read.table("res.genes.Analysis3/global_stats.txt", sep = "\t", header = TRUE, fill = TRUE)
global_stats <- global_stats[order(global_stats$p.value), ]
kable(global_stats[1:min(nrow(global_stats), n_max), ])
```

## "SDC1_BRCA-radiation_therapy-NO"

\begin{center}
\includegraphics[height=250px]{res.genes.Analysis3/SDC1_BRCA-radiation_therapy-NO.png}
\end{center}

## "SDC1_BRCA-person_neoplasm_cancer_status-WITH TUMOR"

\begin{center}
\includegraphics[height=250px]{res.genes.Analysis3/SDC1_BRCA-person_neoplasm_cancer_status-WITH TUMOR.png}
\end{center}

\pagebreak

# Analysis 5: Clinical-centric analysis. Selected cancer, selected clinical subcategory, gene expression differences across categories

Expression of `r gene` in selected clinical subcategories

```{r echo=FALSE, out.height='250px'}
knitr::include_graphics('res.genes.Analysis5/BRCA_SDC1_PAM50Call_RNAseq.png')  # 'PAM50Call_RNAseq' for BRCA, 'subtype' for OV
```

```{r eval=TRUE}
global_stats <- read.table("res.genes.Analysis5/global_stats.txt", sep = "\t", header = TRUE, fill = TRUE)
global_stats <- global_stats[order(global_stats$p.value), c("Cancer", "Gene", "p.value", "HR")]
rownames(global_stats) <- NULL
pander(global_stats)
```

## ANOVA and Tukey's test

What are the means of log2-expression per clinical subgroup""

```{r}
load(file = "res.genes.Analysis5/mtx_to_plot.rda")
tapply(mtx_to_plot$Gene, mtx_to_plot$Clinical, mean)
```

### ANOVA

Is the expression of gene `r gene` significantly different across clinical subgroups? Significant "Pr(>F)" suggests "Yes"

```{r}
# ANOVA
groupdiff <- lm(mtx_to_plot$Gene ~ mtx_to_plot$Clinical)
# summary(groupdiff)
anova(groupdiff)
```

### Tukey HSD (Honest Significant Difference) test

Which pair of clinical categories has significant differences? "p.adj" and confidence intervals that do not cross 0 suggest significant differences in gene expression between the subgroups in the corresponding pairwise comparison.

```{r}
# Tukey HSD
a1 <- aov(mtx_to_plot$Gene ~ mtx_to_plot$Clinical)
# summary(a1)
posthoc <- TukeyHSD(x=a1, 'mtx_to_plot$Clinical', conf.level=0.95)
posthoc %>% pander()
par(las = 2)
par(mar = c(5, 8, 5, 8))
plot(posthoc)
```

\pagebreak 

# Methods

## Survival analysis of gene expression data from TCGA

Level 3 gene expression data summarized as RSEM values was obtained using the `TCGA2STAT` R package v 1.2, along with the corresponding clinical annotations. Data for each of the 34 cancers was obtained separately. The data was log2-transformed and analyzed using Kaplan-Meier curves and Cox proportional hazard model. Each gene of interest was analyzed for its effect on survival by separating patients into high/low expression subgroups. A modified approach from [@Mihaly:2013aa] was used to estimate the best gene expression cutoff that separates high/low expression subgroups with differential survival.

We took the advantage of the availability of clinical annotations. To identify if expression of a gene of interest affects survival in any specific clinical subgroup, subsets of patients annotated with specific clinical annotations were selected (e.g., “males” or “females” in the “gender” clinical annotation). Subgroups with < 40 patients were not considered.

## Differential expression analysis

Samples in the selected cancer cohort were sorted by expression of the selected genes. Differentially expressed genes were detected between samples in the upper 75 percentile of the expression gradient and samples in the lower 25 percentile using `limma` v 3.32.6 R package [@Ritchie:2015aa; @Smyth:2004aa]. P-values were corrected for multiple testing using False Discovery Rate (FDR) method [@Benjamini:1995aa]. Genes differentially expressed at FDR < 0.01 were selected for further analysis.

# References

