### Supplementary Table 2.  Tools for RNA-seq analysis
All tools used in the online tutorial (www.rnaseq.wiki) are referenced below (in bold) along with alternative tools in each category.  Where possible a citation is provided.  Links are also provided to help the user evaluate the code and the level of maintenance.  Where possible the link goes directly to a source controlled repository such as a git repo.  Additional lists of tools can be found here: Alamancos et al. (arXiv), Hooper et al. [24447644], the rna-seqblog, and RNA-seq - Protocols and Algorithms.  This table is meant to be comprehensive but not exhaustive.  Some RNA-seq analysis applications that are not explicitly covered here include co-regulation (co-expression), disease classification, time series, expression compendium databases, outlier expression, data normalization, and miRNA analysis.

| Category | Representative tools |
|----------|:---------------------|
| **Raw data QC** [25577376, 25150838] | FastQC, HTQC [23363224], QC3 [24703969], kPAL [25514851]. |
| **Read trimming** [24376861] | Trimmomatic [24695404], Skewer [24925680], Flexbar [24832523], FASTX. | 
| **Alignment (splice aware, for alignment to a reference genome)** [24185836] | TopHat [19289445, 23618408], STAR [23104886], HISAT [25751142], segemehl [24512684], GSNAP, MapSplice [20802226], JAGuaR [25062255], SpliceMap [20371516], HMMSplicer [21079731], TrueSight/UnSplicer [24259430]. |
| **Alignment (non splice aware for alignment to a reference transcriptome)** [23060614, 23758764] | BowTie [19261174], Bwa [19451168]. |
| **Post-alignment QC** [24185836] | FastQC, samtools [19505943], QuaCRS [25368506], RSeQC [22743226], RNA-SeQC [22539670], Picard CollectRnaSeqMetrics, BAMstats, SAMstat [21088025], BlackOPs [23935067], seqbias [22285831]. |
| **Gene/transcriptome annotation** [24722185, 25319663] | Annocript [25701574], XSAnno [24884593], GeneMark-ET [24990371], WImpiBLAST [24979410], RNASEG [24780064], TSSAR [24674136], Vicinal [24623808], OMIGA [24609470], CoRAL [24145223], AfterParty [24093729], ShortStack [23610128], CIRI [25583365]. |
| **Small RNA identification and characterization (e.g., miRNAs)** [25319663, 23720668] | ShortStack [23610128], CoRAL [24145223], MTide [25256573], FlaiMapper [25338717], miRPlant [25117656], PROmiRNA [23958307], omiRas [23946503], DREAM [25840043]. |
| **Transcript assembly (reference genome guided)** [24185837, 21897427, 23393030] | Cufflinks [20436464], Scripture [20436462], StringTie [25690850], bayesembler [25367074], IsoLasso [21951053]. |
| **Transcript assembly (de novo, reference genome free)** [21897427, 23393030, 23056003, 23666209, 25084827, 25279728, 25788326] | Trinity [23845962], Trans-ABySS [20935650], Oases [22368243], RSEM [21816040], DETONATE [25608678], SEECER (sequencing error correction for assembly) [23558750], BRANCH [23493323] uses partial or related genomics sequences as a guide, EBARDenovo [23457040], Bridger [25723335]. |
| **Transcript abundance or expression estimation (FPKM/RPKM)** [24185837, 24885830, 24109770, 24685233] | Cufflinks [20436464], eXpress [23160280], RSEM [21816040], Sailfish (alignment free) [24752080], RNA-Skim (alignment free) [24931995], MITIE [23980025], ireckon [23204306], DRUT [23202426]. |
| **Obtaining raw transcript/gene read counts (FPM/RPM)** [21176179] | HTSeq [25260700], FeatureCounts [24227677], Rcount [25322836], maxcounts [24564404], FIXSEQ (adjusts counts to compensate for overdispersion) [24603409], Cuffquant. |
| **Differential expression** [25119138, 24300110, 24020486, 25024085] | Cuffdiff [23222703], limma [25605792], DESeq2 [25516281], EdgeR [19910308], Corset (for de novo assembled transcriptomes) [25063469], sSeq [23589650], BADGE [25252852], compcodeR [24813215], metaRNASeq [24678608], Characteristic Direction [24650281], NPEBseq [23981227]. |
| **Alternative splicing, alternative expression** [24447644, 24885830, 24058384, 24549677, 24951248, 25511303] | Cuffdiff [23222703], DEXSeq [22722343], ALEXA-seq [20835245], IUTA [25283306], FineSplice [24574529], PennSeq [24362841], FlipFlop [24813214], SNPlice [25481010], spliceR [24655717], GESS [24447644], RNASeq-MATS [23872975], SplicingCompass [23449093], DiffSplice [23155066], SigFuge [25030904], SUPPA [bioRXiv], CLASS [bioRXiv], SplAdder [bioRXiv], SplicePie [25800735]. |
| **Variant (e.g., SNP) and mutation detection** [23555596, 24075185, 22468815], germline or somatic, and eQTL/sQTL characterization [25733796] | GATK (Best Practices Guide) [20644199], samtools [19505943], SNVMix [20130035], SNPlice [25481010], eSNV-detect [25352556], RVboost [25170027], sQTLseekeR [25140736], eQTL/ASE,  [BioRXiv], SNiPloid [24163691], SNPiR [24075185], QualitySNPng [23632165], RNAmapper [23299976], CRAC [23537109], RADIA [25405470]. |
| **RNA editing** [22327324, 23291724, 23598527, 25859542] | REDItools [23742983], GIREMI [25730491], ICEBreaker [25855956]. |
| **Allele specific expression** [23919664, 25183311, 25339465] | AlleleSeq [21811232], Allim [23615333], mamba [25819081], EMASE, MBASED [25315065], limma [25605792].
| **Viral detection** [23740984, 23279287, 22647373] | VirusSeq [23162058], VirusFinder [23717618], RNA CoMPASS [24586784]. | 
| **Fusion detection** [25500544, 25266161, 23815381, 23555082, 25286921] | FusionQ [23815381], TRUP [25650807], Dissect [22689759], Trans-ABySS [20935650], PRADA (RNA-seq pipeline with a fusion module) [24695405], Pegasus (used for fusion annotation) [25183062], FusionCatcher, ChimeraScan [21840877], TopHat-fusion [21835007], BreakFusion [22563071], deFuse [21625565], FusionHunter [21546395], EricScript [23093608], Barnacle [23941359], bellerophontes [22711792], Chimera (merge results from multiple fusion algorithms) [25286921], GFML (format for representing fusion data) [23072312]. | 
| **Visualization** [24792048, 25757788] | SplicingViewer [22226708], IGV [22517427], Sashimi plots [25617416], IGB (splicing visualization protocol) [24792048], PrimerSeq (Visualize RNA-seq data for primer design) [24747190], ASTALAVISTA [25577392], Circos [19541911], Epiviz [25086505], RNAbrowse [24823498], ZENBU [24727769], RNAseqViewer [24215023], viRome [23709497], miRseqViewer [25322835], Circleator [25075113], RNASeqBrowser [25766521]. |
| **Integration of DNA-seq and RNA-seq data** [23499923] | Veridical [24741438], SpliceFinder [24498620], nFuse [22745232], RADIA [25405470]. |





